test/FIFOMarket.test.ts

```diff
@@ -678,20 +710,21 @@ describe('FIFOMarket', () => {
     });
     it('should mint a certificate with multiple removals per supplier in round robin order and update state variables', async () => {
       const buyerInitialBPNoriBalance = formatTokenAmount(1_000_000);
-      const numberOfRemovalsToCreate = 100;
+      const amountOfRemovals = 100;
+      const amountPerRemoval = 50;
       const removalDataToList = [
-        ...Array.from({ length: numberOfRemovalsToCreate }).keys(),
-      ].map((_) => {
-        return { amount: 50 };
+        ...Array.from({ length: amountOfRemovals }).keys(),
+      ].map((_, index) => {
+        return { amount: amountPerRemoval, vintage: 2015 + (index % 5) };
       });
       const { bpNori, certificate, fifoMarket, hre } = await setupTestLocal({
         buyerInitialBPNoriBalance,
         removalDataToList,
       });
       const { supplier, buyer, noriWallet } = hre.namedAccounts;
 
-      const purchaseAmount = '5000'; // purchase all supply
-      const fee = '750';
+      const purchaseAmount = (amountPerRemoval * amountOfRemovals).toString(); // purchase all supply
```

Comment:
should we just make this the following?:

```suggestion
      const purchaseAmount = (amountPerRemoval * await fifoMarket.totalNumberActiveRemovals()).toString(); // purchase all supply
```

Reply:
I don't think so, since we're trying to buy the full amount of removals it makes more sense to use the input data as our counter, just in case something goes silently wrong when listing removals (and avoids an unnecessary await). But that's just personal preference.

Reply:
makes more sense to calculate it explicitly from `removalDataToList` then

Reply:
I kinda disagree, since `removalDataToList` is derived from the `amountOfRemovals` in the first place. I think using that initially defined variable establishes the clearest intent for this particular function of the test. Can you help me understand what issues you're seeing with this?

Reply:
I see, fair enough, will change